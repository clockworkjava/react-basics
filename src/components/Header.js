import React from 'react';
import { Link } from 'react-router-dom'

class Header extends React.Component {

    constructor() {
        super();
        this.state = {
            bookstoreName : "Black Books",
            clicked : true,
            textColor : "white",
            backgroundColor : "black"
        }
    }

    handleClick = () => {

        if(this.state.clicked) {
            this.setState({
                bookstoreName : "White Books",
                textColor : "black",
                backgroundColor : "white"
            })
        } else {
            this.setState({
                bookstoreName : "Black Books",
                textColor : "white",
                backgroundColor : "black"
             
            })
        }
        this.setState({
            clicked : !this.state.clicked
        })
    }

    render() {

        let headerCss = {
            color : this.state.textColor,
            backgroundColor : this.state.backgroundColor
        }

        return (
            <div className="row header" style={headerCss} onClick={this.handleClick}>
                <center><h1>{this.state.bookstoreName}</h1></center>
                <Link to="/admin"><button className="btn btn-info goToAdmin">Administrator Panel</button></Link>
            </div>
        );
    }
}

export default Header;