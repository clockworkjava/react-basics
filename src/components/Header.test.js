import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Header tests', () => {

    it('Header renders without a problem', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Header changes text according to state', () => {
        const wrapper = shallow(<Header/>);
        // console.log(wrapper.debug());
        expect(wrapper.find('h1').text()).toBe('Black Books');
        wrapper.setState({
            bookstoreName: "White Books"
        });
        expect(wrapper.find('h1').text()).toBe('White Books');
    })

    it('Header state changes after clicking on header div', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper.state().bookstoreName).toBe('Black Books');
        wrapper.find('.header').simulate('click');
        expect(wrapper.state().bookstoreName).toBe('White Books');
    })

    it('Snapshot matches', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    })

})

