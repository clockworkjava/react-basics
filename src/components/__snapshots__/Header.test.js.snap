// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Snapshot matches 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <center>
        <h1>
                Black Books
        </h1>
</center>,
      "className": "row header",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "black",
        "color": "white",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Black Books
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Black Books",
        },
        "ref": null,
        "rendered": "Black Books",
        "type": "h1",
      },
      "type": "center",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <center>
          <h1>
                    Black Books
          </h1>
</center>,
        "className": "row header",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "black",
          "color": "white",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Black Books
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Black Books",
          },
          "ref": null,
          "rendered": "Black Books",
          "type": "h1",
        },
        "type": "center",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
